### ISSUE TITLE
Add ability to customize the Content-Transfer-Encoding header

### ISSUE THREAD

**Issue by soywod** [stalwartlabs/mail-builder#20]
When creating encrypted multipart, the 2 inner parts `application/pgp-encrypted` and `application/octet-stream` should not contain anything else. From the [RFC](https://www.rfc-editor.org/rfc/rfc3156#section-4):

> The multipart/encrypted MIME body MUST consist of exactly two body
> parts, the first with content type "application/pgp-encrypted".  This
> body contains the control information.  A message complying with this
> standard MUST contain a "Version: 1" field in this body.  **Since the
> OpenPGP packet format contains all other information necessary for
> decrypting, no other information is required here.**
> 
> The second MIME body part MUST contain the actual encrypted data.  It
> MUST be labeled with a content type of "application/octet-stream".

The first inner part that should contain `Version: 1` ends up being base64 encoded:

```eml
Content-Type: application/pgp-encrypted
Content-Transfer-Encoding: base64

VmVyc2lvbjogMQ==

```

Instead of:

```eml
Content-Type: application/pgp-encrypted

Version: 1

```

I guess adding a condition on encrypted (and signed) parts could be enough. Or adding the ability to override (or remove) the header.

**Comment by mdecimus** [comment]
Please check the new version I've just committed to this repo. It is now much simpler and less confusing to add message parts.
With the new version make sure you pass the pgp-encrypted contents as a string so they are not base64 encoded:

```
Multipart::new("application/pgp-encrypted", "Version:1 \r\n");
```

**Comment by soywod** [comment]
It works way better, and the API is indeed simpler to use. Thank you!

**Comment by mdecimus** [comment]
Great, I'll do a few tests and publish the new version to crates.io next week.

**Comment by mdecimus** [comment]
@soywod I tested the new version and it looks fine, so I've just published mail-builder 0.3 on crates.io.
